apollo-router/tests/integration_tests.rs

```diff
@@ -540,9 +589,22 @@ async fn normal_query_with_defer_accept_header() {
         .header(ACCEPT, "multipart/mixed; deferSpec=20220824")
         .build()
         .expect("expecting valid request");
-    let (actual, _registry) = query_rust_with_config(request, serde_json::json!({})).await;
+    let (actual, _registry) = {
+        let (router, counting_registry) = setup_router_and_registry(serde_json::json!({})).await;
+        (
+            router
+                .oneshot(request.try_into().unwrap())
+                .await
+                .unwrap()
+                .next_response()
+                .await
+                .unwrap()
+                .unwrap(),
+            counting_registry,
+        )
+    };
 
-    assert!(actual.errors.is_empty());
+    insta::assert_snapshot!(std::str::from_utf8(actual.to_vec().as_slice()).unwrap());
```

Comment:
```suggestion
    insta::assert_snapshot!(serde_json::from_slice::<serde_json::Value>(&actual).unwrap());
```

This will conserve the snapshot shape for all integration tests

Reply:
i m going to update this before the mob review

Reply:
Thanks, that will simplify the PR further

Reply:
I lied, but i ll update it asap

Reply:
fixed (in an elegant way i think :D) in 9c5f4908